data_psi.go

```diff
@@ -462,18 +462,42 @@ func (d *PSIData) toData(firstPacket *Packet, pid uint16) (ds []*DemuxerData) {
 		// Switch on table type
 		switch s.Header.TableID {
 		case PSITableIDNITVariant1, PSITableIDNITVariant2:
-			ds = append(ds, &DemuxerData{FirstPacket: firstPacket, NIT: s.Syntax.Data.NIT, PID: pid})
+			var nit *NITData
+			if s.Syntax != nil {
```

Comment:
Could add the following `if` statement before the `switch` instead ?

```go
// No data
if s.Syntax == nil || s.Syntax.Data == nil {
    return
}
```

Reply:
Done.